### ISSUE TITLE
WRC (1849250)

### ISSUE THREAD

**Issue by Lawstorant** [ValveSoftware/Proton#7220]
# Compatibility Report
- Name of the game with compatibility issues: EA Sports WRC
- Steam AppID of the game: 1849250

## System Information
- Steam Deck and PC:
- GPU: RX 6800XT
- Video driver version: Mesa 23.2
- Kernel version: 6.1-LTS
- Proton version: 8.0-4

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

## Symptoms
The game works fine on Linux with Proton but it's blocked on Steam Deck with "Running on Steam Deck is not supported." error. We need a way to circumvent this artificial block. The same thing happens when running steam deck client on PC.

## Reproduction
1. Launch EA Sports WRC.
2. Application shows Error when it detects Steam Deck client

**Comment by Samsagax** [comment]
Long shot but maybe worth a try. Have you tested setting `SteamDeck=0 %command%` as launch options?

**Comment by ashtophet** [comment]
> Long shot but maybe worth a try. Have you tested setting `SteamDeck=0 %command%` as launch options?

It works ;)

**Comment by Lawstorant** [comment]
Yup. SteamDeck=0 works on my Deck and PC.

**Comment by TheAquabat** [comment]
I dont have a steamdeck, but with a 6700XT I am experiencing something weird regarding performance. Sometimes the game will run perfectly fine hitting about 70-80 fps, and sometimes I will launch it and will hit 30fps at max. I tried pinning the cpu cores but didnt work.

**Comment by Samsagax** [comment]
According to the devs, a fix for stuttering is coming. The same has been reported on Windows.
My fix for now has been to find a balance with Textures. I play at Ultra but lowered Textures to High until those dips are unnoticeable (not gone, but won't bother you).

**Comment by TheAquabat** [comment]
this is not stuttering, this is something else, its not just performance temporally dipping. Sometimes when I run the game it will not go over 30 fps, peformance will take a heavy dent, its like fps are suddenly capped idk, and no matter what I do, or setting I change nothing fixes it, I'd expect 70 fps or a tad more to be more in line with my hardware.

EDIT: some windows users are reporting a similar issue like me, like having 30 fps on a 3080, makes no sense. For me its not playable on some stages I am hitting 20-15 fps (not lowest peak, but on average).

**Comment by TheAquabat** [comment]
this bug is back
```
[  187.840177] umip: WRC.exe[7089] ip:1540c03bd sp:b976f0: SGDT instruction cannot be used by applications.
[  187.840182] umip: WRC.exe[7089] ip:1540c03bd sp:b976f0: For now, expensive software emulation returns the result.
[  193.905940] umip: WRC.exe[7089] ip:15082a152 sp:b9df38: SGDT instruction cannot be used by applications.
[  193.905944] umip: WRC.exe[7089] ip:15082a152 sp:b9df38: For now, expensive software emulation returns the result.
[  193.986632] umip: WRC.exe[7089] ip:14bd6e6be sp:b9de38: SGDT instruction cannot be used by applications.
```

**Comment by Lawstorant** [comment]
Yes, the game has multiple performance issues. There's a huge CPU bottleneck where the game is often putting too much load on a single core. With my 6800XT I get the same framerate be it 1080p, 1440p or 4k

**Comment by xpander69** [comment]
So far i haven't encountered any issues with framerates dropping or any of the mentioned CPU bottlenecks. Seems to run 70-120 FPS for me with few minor shader related stutters on the first time played stages, but nothing major.
2560x1440, High settings, DLSS quality, Motion Blur off, Textures Ultra.

However i see this in my dmesg a lot:
```
[Sat Nov  4 19:38:04 2023] umip_printk: 43 callbacks suppressed
[Sat Nov  4 19:38:04 2023] umip: WRC.exe[2764771] ip:15409c2eb sp:b92908: SGDT instruction cannot be used by applications.
[Sat Nov  4 19:38:04 2023] umip: WRC.exe[2764771] ip:15409c2eb sp:b92908: For now, expensive software emulation returns the result.
[Sat Nov  4 19:38:04 2023] umip: WRC.exe[2764771] ip:15409c2eb sp:b9ab08: SGDT instruction cannot be used by applications.
[Sat Nov  4 19:38:04 2023] umip: WRC.exe[2764771] ip:15409c2eb sp:b9ab08: For now, expensive software emulation returns the result.
```
The game spams that like crazy. Though that doesn't seem to affect performance.

PC specs:
CPU: Ryzen 7 5800X3D
GPU: nvidia RTX 3080
Video driver version: 545.29.02
Kernel version: 6.5-pds
Proton version: proton-experimental

**Comment by TheAquabat** [comment]
> Yes, the game has multiple performance issues. There's a huge CPU bottleneck where the game is often putting too much load on a single core. With my 6800XT I get the same framerate be it 1080p, 1440p or 4k

I wonder if this is related to some type of foliage, particles or sceneries ? on Japan at night I got 40 fps on average for example.
Maybe there is several problems going on here.
Lets see if the upcoming performance patch fixes this.

**Comment by xpander69** [comment]
Have driven couple of japan stages at night now also and no issues + several other locations in night and other conditions. Fps seems to be always about same. Dropping down to 70-80 where lots of crowd is next to the track and 90-100 in more open field/forest areas.
Its very strange then. Is it somehow related to AMD GPUs as im using Nvidia. Or its the CPU differences or something else.

edit: there was a game update now and also proton-experimental was updated

messages like
`umip: WRC.exe[2764771] ip:15409c2eb sp:b92908: For now, expensive software emulation returns the result.`
seem to have disappeared also. No idea if game update fixed it or proton one.
Game still works great as before.

edit2: Ohh i'm stupid i forgot that i added `clearcpuid=514` boot parameter. That explains why those dmesg messages dont show up anymore.

**Comment by TheAquabat** [comment]
With the latest patch I think that most of my performance issues have gone.

**Comment by kisak-valve** [comment]
EA SPORTS™ WRC: Problems with force feedback

Issue transferred from https://github.com/ValveSoftware/Proton/issues/7370.
@eruizc-dev posted on 2023-12-22T17:24:27:

# Compatibility Report

- Name of the game with compatibility issues: EA SPORTS™ WRC
- Steam AppID of the game: 1849250

## System Information

- GPU: RX 6650XT
- Video driver version: Mesa 23.3.0
- Kernel version: 6.6.7-zen1-1-zen
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/eruizc-dev/9b88f865cd4b0347134d342b6dbe747e
- Proton version: tried both 7.0-6 and 8.0-4
- Wheel: Logitech Trueforce G923 - Play Station edition 

## I confirm:

- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.



## Symptoms 

When using a Logitech wheel and force feedback settings are set to default, the wheel will start going left and right non stop after you turn it with a bit of force.

https://github.com/berarma/new-lg4ff/assets/58179604/30a6962d-9cd9-411e-9f38-4dc052f7b788

A workaround to this problem is to set tire friction settings to a low value (between 0 and 25)

## Reproduction

 1. Open the game
 2. Go into a stage with the default control settings
 3. Spin your wheel fast to either side
 4. It should start going left and right non-stop like in the video
***
@eruizc-dev commented on 2023-12-22T17:24:45:

Also reported in https://github.com/berarma/new-lg4ff/issues/86

**Comment by Faceless3882** [comment]
Any update on the force feedback issue with Logitech Trueforce G923 - Play Station edition? I'm having the exact same problem. Super annoying

**Comment by eruizc-dev** [comment]
@Faceless3882 have you tried the workaround setting tire friction to some low value like 15-20?

**Comment by Faceless3882** [comment]
> @Faceless3882 have you tried the workaround setting tire friction to some low value like 15-20?

Yes, it gets around it and makes it playable but it ruins the feel of the cars

**Comment by Samsagax** [comment]
Too bad. This game is done. Since they announced EA kernel level Anticheat is coming in the next version, then we are basically screwed. EA doesn't care and they stated it so in their press release.

**Comment by TheAquabat** [comment]
> Too bad. This game is done. Since they announced EA kernel level Anticheat is coming in the next version, then we are basically screwed. EA doesn't care and they stated it so in their press release.

got me thinking.. isnt there an option to cancel going online, tho? because you could downgrade the game with app_depot previous version and then play offline? isnt that possible?

**Comment by Samsagax** [comment]
> got me thinking.. isnt there an option to cancel going online, tho? because you could downgrade the game with app_depot previous version and then play offline? isnt that possible?

 One of the good things about this game is clubs. Playing online with friends and such.

**Comment by solenum** [comment]
Officially borked with the [latest patch](https://www.ea.com/games/ea-sports-wrc/wrc/news/ea-anticheat?utm_campaign=wrc23_hd_ww_ic_soco_blog_wrc-anticheat&utm_source=blog&utm_medium=social&cid=80350&ts=1721771686020) that introduces anticheat as of a week or so ago

> Additionally, EA SPORTS™ WRC will not run on the native Steam OS for Steam Deck following the release of EA anticheat.

Edit;  For anyone coming back here to check if there's been any progress - EA effectively killed off codemasters and this game, and there's unlikely to be any changes that allow it to run under Proton.
